### ISSUE TITLE
Module Import Errors When Training with Custom Dataset

### ISSUE THREAD

**Issue by tomermesser** [facebookresearch/UHM#6]
Hello,

I'm attempting to train the model using my own data following the "Training and testing with a short phone scan" instructions provided in the repository's README. However, I consistently encounter module import errors that prevent the training scripts from running successfully. Below, I've detailed my setup, the steps I've taken, and the errors I'm facing.

```
/content/UHM
./
├── Avatar/
│   ├── main/
│   │   └── config.py - **Got modified***
│   │   └── run.py - **Got modified***
├── UHM/
│   └── demo/
│       └── ckpt.pth
│       └── template.obj
│       └── template.pkl
│       └── uhm.py
├── data/
│   └── Custom/
│       └── data/
│           └── YourSubjectName/
│               └── 0/
│                   ├── mano_params/
│                   │   ├── params/
│                   │   ├── renders/
│                   │   └── meshes/
│                   ├── cam_param.json
│                   ├── masks/
│                   ├── depthmaps/
│                   ├── keypoints/
│                   └── frames/
│               └── frame_list_test.txt
│               └── frame_list_train.txt

```

**Steps Taken**:

A. I modified the /content/UHM/Avatar/main/config.py file as follows (lines 23-27):
```
## Others
dataset = 'Custom'  # 'Custom' to use my own dataset
num_thread = 16
num_gpus = 1
continue_train = False
```

B. I modified the /content/UHM/Avatar/main/run.py file as follows (lines 13):
```
subject_id_list = ['YourSubjectName'] # 
```

C. I enter the folder - /content/UHM/Avatar/main
D. I ran the command:
`!python run.py`

And I got those errors, 
```
python train.py --subject_id YourSubjectName
Traceback (most recent call last):
  File "/content/UHM/Avatar/main/train.py", line 12, in <module>
    from base import Trainer
  File "/content/UHM/Avatar/main/../common/base.py", line 21, in <module>
    from model import get_model
  File "/content/UHM/Avatar/main/model.py", line 12, in <module>
    from nets.layer import SoftMaskRenderer, DepthmapRenderer, UVRenderer, XY2UV
  File "/content/UHM/Avatar/main/../common/nets/layer.py", line 12, in <module>
    from utils.uhm import uhm
  File "/content/UHM/Avatar/main/../common/utils/uhm.py", line 11, in <module>
    from utils.UHM.uhm import UHM
ModuleNotFoundError: No module named 'utils.UHM'
python test.py --subject_id YourSubjectName --test_epoch 99
Traceback (most recent call last):
  File "/content/UHM/Avatar/main/test.py", line 16, in <module>
    from base import Tester
  File "/content/UHM/Avatar/main/../common/base.py", line 21, in <module>
    from model import get_model
  File "/content/UHM/Avatar/main/model.py", line 12, in <module>
    from nets.layer import SoftMaskRenderer, DepthmapRenderer, UVRenderer, XY2UV
  File "/content/UHM/Avatar/main/../common/nets/layer.py", line 12, in <module>
    from utils.uhm import uhm
  File "/content/UHM/Avatar/main/../common/utils/uhm.py", line 11, in <module>
    from utils.UHM.uhm import UHM
ModuleNotFoundError: No module named 'utils.UHM'
python train.py --subject_id YourSubjectName --use_tex --continue
Traceback (most recent call last):
  File "/content/UHM/Avatar/main/train.py", line 12, in <module>
    from base import Trainer
  File "/content/UHM/Avatar/main/../common/base.py", line 21, in <module>
    from model import get_model
  File "/content/UHM/Avatar/main/model.py", line 12, in <module>
    from nets.layer import SoftMaskRenderer, DepthmapRenderer, UVRenderer, XY2UV
  File "/content/UHM/Avatar/main/../common/nets/layer.py", line 12, in <module>
    from utils.uhm import uhm
  File "/content/UHM/Avatar/main/../common/utils/uhm.py", line 11, in <module>
    from utils.UHM.uhm import UHM
ModuleNotFoundError: No module named 'utils.UHM'
python test.py --subject_id YourSubjectName --use_tex --test_epoch 149
Traceback (most recent call last):
  File "/content/UHM/Avatar/main/test.py", line 16, in <module>
    from base import Tester
  File "/content/UHM/Avatar/main/../common/base.py", line 21, in <module>
    from model import get_model
  File "/content/UHM/Avatar/main/model.py", line 12, in <module>
    from nets.layer import SoftMaskRenderer, DepthmapRenderer, UVRenderer, XY2UV
  File "/content/UHM/Avatar/main/../common/nets/layer.py", line 12, in <module>
    from utils.uhm import uhm
  File "/content/UHM/Avatar/main/../common/utils/uhm.py", line 11, in <module>
    from utils.UHM.uhm import UHM
ModuleNotFoundError: No module named 'utils.UHM'
python get_id.py --subject_id YourSubjectName --test_epoch 149
Traceback (most recent call last):
  File "/content/UHM/Avatar/main/get_id.py", line 12, in <module>
    from base import Tester
  File "/content/UHM/Avatar/main/../common/base.py", line 21, in <module>
    from model import get_model
  File "/content/UHM/Avatar/main/model.py", line 12, in <module>
    from nets.layer import SoftMaskRenderer, DepthmapRenderer, UVRenderer, XY2UV
  File "/content/UHM/Avatar/main/../common/nets/layer.py", line 12, in <module>
    from utils.uhm import uhm
  File "/content/UHM/Avatar/main/../common/utils/uhm.py", line 11, in <module>
    from utils.UHM.uhm import UHM
ModuleNotFoundError: No module named 'utils.UHM'
Traceback (most recent call last):
  File "/content/UHM/Avatar/main/run.py", line 42, in <module>
    model = torch.load(model_path)
  File "/usr/local/lib/python3.10/dist-packages/torch/serialization.py", line 791, in load
    with _open_file_like(f, 'rb') as opened_file:
  File "/usr/local/lib/python3.10/dist-packages/torch/serialization.py", line 271, in _open_file_like
    return _open_file(name_or_buffer, mode)
  File "/usr/local/lib/python3.10/dist-packages/torch/serialization.py", line 252, in __init__
    super().__init__(open(name, mode))
FileNotFoundError: [Errno 2] No such file or directory: '../output/model_dump/YourSubjectName/snapshot_149.pth'
```

What I missed along the process ? 
I want to achieve better results for my video/3D model. 

Any help will be much appreciate. 
Thanks !

**Comment by mks0601** [comment]
Hi, your issue is too hard to read. You can follow suggested directory without modifying it and try again.

**Comment by tomermesser** [comment]
@mks0601 thanks for the quick replay ! 
I rewrite my issue, hope it is more understood now. 
[](https://github.com/facebookresearch/UHM/issues/6#issue-2640902369)

**Comment by tomermesser** [comment]
Hi @mks0601 

I've successfully set up the project, and I must say, it's quite impressive! Initially, I encountered some issues with data organization. However, after arranging my files properly under the /content/UHM/Avatar/data/Custom/data directory, everything worked seamlessly.

I'm reaching out to explore potential improvements to the results I'm currently achieving. Any advice or recommendations on how to enhance the output would be greatly appreciated!

Thank you for creating such a fantastic repository.

3D MESH:
https://drive.google.com/file/d/1F7uiEmnD7Hti699ByCuflvn7UpCgws7l/view?usp=sharing
Final Video:
https://drive.google.com/file/d/1DCvcQtuTGREAa8qDiaP3vRerKAISoFzK/view?usp=sharing
